### PR TITLE
Bump `isort` version to 5.6

### PR DESCRIPTION
**Pull request by charlesbluca**
This change is intended for 21.08

[With `isort` 5.6.0 came the ability to resort .pxd files](https://pycqa.github.io/isort/CHANGELOG/#560-october-7-2020), in addition to .pyx. By bumping the version here, we could begin enforcing .pxd package sorting on RAPIDS projects with Cython code.

Also relevant to rapidsai/cudf#8215

### CODE REVIEW

**Review by ajschmidt8 — APPROVED** [review]

**Comment by jakirkham** [comment]
JFYI this is still targeting `branch-21.06` as `branch-21.08` wasn't available when this was opened. We are happy to move over to `branch-21.08` once it is ready 🙂

**Comment by ajschmidt8** [comment]
Converted this PR to a draft so that it doesn't accidentally get merged to `21.06`.

**Comment by charlesbluca** [comment]
Is this safe to mark as ready for review?

**Comment by jakirkham** [comment]
Yes it's been retargeted to the new branch

**Comment by charlesbluca** [comment]
Trying to assert if this will be a breaking PR, i.e. if code resorted by `isort=5.0` will pass checks with `isort=5.6`; depending on this, we might want to open follow up PRs with relevant Python repos to make sure the version in their pre-commit hooks is also bumped

**Comment by ajschmidt8** [comment]
> Trying to assert if this will be a breaking PR, i.e. if code resorted by `isort=5.0` will pass checks with `isort=5.6`; depending on this, we might want to open follow up PRs with relevant Python repos to make sure the version in their pre-commit hooks is also bumped

@charlesbluca, sounds like we might want to figure that out before merging this. Please post your findings here. Until then, I'll switch this to a draft so we don't break anything with an accidental merge.

**Comment by charlesbluca** [comment]
It looks like the repos this could potentially break are those using pre-commit that have Cython code:

- cuspatial
- cudf (rapidsai/cudf#8215 already addressing this)
- ucx-py
- rmm
- rmm-testbed

Essentially, we would either want to make sure these repos' gpuCI `isort` checks are ignoring Cython code, or that their `isort` pre-commit hooks are resorting it and this is being enforced by gpuCI.

**Comment by charlesbluca** [comment]
Opened up PRs for the impacted repos:

- rapidsai/cuspatial#419
- rapidsai/rmm#806
- rapidsai/ucx-py#753

I skipped over rmm-testbed as it seems like that repo would probably pull the changes from rmm, but happy to open up a PR there too if that isn't the case

**Comment by ajschmidt8** [comment]
> Opened up PRs for the impacted repos:
> 
> * [rapidsai/cuspatial#419](https://github.com/rapidsai/cuspatial/pull/419)
> * [rapidsai/rmm#806](https://github.com/rapidsai/rmm/pull/806)
> * [rapidsai/ucx-py#753](https://github.com/rapidsai/ucx-py/pull/753)
> 
> I skipped over rmm-testbed as it seems like that repo would probably pull the changes from rmm, but happy to open up a PR there too if that isn't the case

@charlesbluca, `rmm-testbed` is fine to skip :+1:. can you review the status of these PRs? I see a few failures. It seems some may be related to `isort`. I'm not sure if they'll be fixed after this dependency is updated. If everything is good to go here, I can start approving/merging all these PRs.

**Comment by charlesbluca** [comment]
Sure! I'd imagine the failures are because `isort` is now checking the PYX files and failing because they are not sorted properly - ideally, what we would want to do is run the `isort=5.6` hook on all the codebases in the commit as well, which should resolve any failures provided that there aren't any sorting quirks within the Cython files.

I can do this now, which should request review from the Python codeowners of the respective repos.

**Comment by charlesbluca** [comment]
With all the Cython PRs merged, this should be safe to merge now

**Review by Ethyling — APPROVED** [review]

**Comment by galipremsagar** [comment]
rerun tests

**Pull request merged**
